File: apps/sdkwork-rtc-h5/src/bootstrap/adminServices.ts
import {
  MediaArtifactService,
  MediaSessionService,
  ProviderAccountService,
  ProviderApplicationService,
  ProviderCredentialService,
  ProviderPluginService,
  ProviderProfileService,
  ProviderQueryJobService,
  ProviderRouteService,
  ProviderSchemaService,
  ProviderWebhookService,
  QualitySampleService,
  RoomService,
  createBackendRtcClient,
} from "@sdkwork/rtc-h5-admin-core";

import {
  bootstrapAdminAuth,
  buildAdminSdkHeaders,
  loadAdminSession,
} from "./adminAuth";
import { resolveEnvironment } from "./environment";
import { getTokenManager } from "./tokenManager";

export interface RtcAdminServices {
  accounts: ProviderAccountService;
  applications: ProviderApplicationService;
  credentials: ProviderCredentialService;
  profiles: ProviderProfileService;
  routes: ProviderRouteService;
  schemas: ProviderSchemaService;
  rooms: RoomService;
  mediaSessions: MediaSessionService;
  mediaArtifacts: MediaArtifactService;
  qualitySamples: QualitySampleService;
  plugins: ProviderPluginService;
  webhooks: ProviderWebhookService;
  queryJobs: ProviderQueryJobService;
}

export function createAdminServices(): RtcAdminServices {
  const { backendApiBaseUrl } = resolveEnvironment();
  const session = bootstrapAdminAuth() ?? loadAdminSession();
  const tokenManager = getTokenManager();
  const clientOptions = session
    ? {
        tokenManager,
        authToken: session.authToken,
        accessToken: session.accessToken,
        tenantId: session.tenantId,
        organizationId: session.organizationId,
        headers: buildAdminSdkHeaders(session),
      }
    : { tokenManager };

  const client = createBackendRtcClient(backendApiBaseUrl, clientOptions);

  return {
    accounts: new ProviderAccountService(client),
    applications: new ProviderApplicationService(client),
    credentials: new ProviderCredentialService(client),
    profiles: new ProviderProfileService(client),
    routes: new ProviderRouteService(client),
    schemas: new ProviderSchemaService(client),
    rooms: new RoomService(client),
    mediaSessions: new MediaSessionService(client),
    mediaArtifacts: new MediaArtifactService(client),
    qualitySamples: new QualitySampleService(client),
    plugins: new ProviderPluginService(client),
    webhooks: new ProviderWebhookService(client),
    queryJobs: new ProviderQueryJobService(client),
  };
}

File: apps/sdkwork-rtc-flutter-mobile/lib/bootstrap/admin_services.dart
import 'package:sdkwork_rtc_flutter_mobile_admin_core/sdkwork_rtc_flutter_mobile_admin_core.dart';

import 'admin_auth.dart';
import 'environment.dart';

class RtcAdminServices {
  final ProviderAccountService accounts;
  final ProviderApplicationService applications;
  final ProviderCredentialService credentials;
  final ProviderProfileService profiles;
  final ProviderRouteService routes;
  final ProviderSchemaService schemas;
  final RoomService rooms;
  final ProviderPluginService plugins;
  final ProviderWebhookService webhooks;
  final ProviderQueryJobService queryJobs;

  const RtcAdminServices({
    required this.accounts,
    required this.applications,
    required this.credentials,
    required this.profiles,
    required this.routes,
    required this.schemas,
    required this.rooms,
    required this.plugins,
    required this.webhooks,
    required this.queryJobs,
  });
}

RtcAdminServices createAdminServices({required RtcAdminSession session}) {
  final env = resolveEnvironment();
  final bundle = createRtcBackendClient(
    backendApiBaseUrl: env.backendApiBaseUrl,
    accessToken: session.accessToken,
    authToken: session.authToken,
    tenantId: session.tenantId,
    organizationId: session.organizationId,
    userId: session.userId,
    permissionScope: defaultAdminPermissionScope,
  );
  final client = bundle.backendSdk;

  return RtcAdminServices(
    accounts: ProviderAccountService(client),
    applications: ProviderApplicationService(client),
    credentials: ProviderCredentialService(client),
    profiles: ProviderProfileService(client),
    routes: ProviderRouteService(client),
    schemas: ProviderSchemaService(client),
    rooms: RoomService(client),
    plugins: ProviderPluginService(client),
    webhooks: ProviderWebhookService(client),
    queryJobs: ProviderQueryJobService(client),
  );
}

PersistProviderWizardServices toWizardServices(RtcAdminServices services) {
  return (
    accounts: services.accounts,
    applications: services.applications,
    credentials: services.credentials,
    profiles: services.profiles,
  );
}
